webdriver/tests/bidi/browsing_context/set_viewport/set_device_pixel_ratio.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+
+from ... import get_device_pixel_ratio, get_viewport_dimensions
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("device_pixel_ratio", [1, 2])
+async def test_set_device_pixel_ratio(bidi_session, new_tab, device_pixel_ratio):
```

Comment:
> Also it might be good to have a test as well that uses a higher DPR as the currently used device.

That would not be supported by WebKit so I have not added it.

Reply:
Floats are a bit more difficult to assert due to rounding and I am not sure 0.1 is a use case that is important. Let's add it separately.